File: epi_judge_python/swap_bits.py
from test_framework import generic_test


def swap_bits(x, i, j):
    if i == j or ((x >> i) & 1 == (x >> j) & 1):
        return x

    x ^= (1 << i)
    x ^= (1 << j)
    return x


if __name__ == '__main__':
    exit(
        generic_test.generic_test_main('swap_bits.py', 'swap_bits.tsv',
                                       swap_bits))

File: epi_judge_java/epi/SwapBits.java
package epi;

import epi.test_framework.EpiTest;
import epi.test_framework.GenericTest;

public class SwapBits {

    @EpiTest(testDataFile = "swap_bits.tsv")
    public static long swapBits(long x, int i, int j) {
        if (((x >> i) & 1) == ((x >> j) & 1)) return x;
        long mask = (1L << i) | (1L << j);
        return x ^ mask;
    }

    public static void main(String[] args) {
        System.exit(
                GenericTest
                        .runFromAnnotations(args, "SwapBits.java",
                                new Object() {
                                }.getClass().getEnclosingClass())
                        .ordinal());
    }
}
